packages/ag-charts-enterprise/src/features/annotations/scenes/shapes.ts

```diff
@@ -0,0 +1,92 @@
+import { _ModuleSupport, _Scene } from 'ag-charts-community';
+
+const magnitude = (x: number, y: number) => Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
+
+export class CollidableLine extends _Scene.Line {
```

Comment:
I'll probably merge this into the `_Scene.Line` class under a flag later if this works out.

A visual may help explain it better:


https://github.com/ag-grid/ag-charts/assets/3817697/a6f44fbb-f41e-430e-9dce-62b2fe4a4bfc